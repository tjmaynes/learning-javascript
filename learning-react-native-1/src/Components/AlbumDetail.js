import React from 'react';
import { Image, View, Text, Linking } from 'react-native';

import Card from './Card';
import CardSection from './CardSection';
import CardMetadata from './CardMetadata';

import Button from './Button';

const AlbumDetail = ({ album }) => {
  return (
    <Card>
      <CardSection>
	<CardMetadata album={album} />
      </CardSection>

      <CardSection>
	<Image
	  source={{ uri: album.image }}
	  style={ styles.imageStyle } />
      </CardSection>

      <CardSection>
	<Button onPress={() => Linking.openURL(album.url)}>
	  Buy Now
	</Button>
      </CardSection>
    </Card>
  );
};

const styles = {
  imageStyle: {
    height: 300,
    flex: 1,
    width: null
  }
};

export default AlbumDetail;
