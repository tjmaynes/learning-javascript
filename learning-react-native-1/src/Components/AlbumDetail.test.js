import React from 'react';
import AlbumDetail from './AlbumDetail';

import renderer from 'react-test-renderer';

const albumStub = [
  {
    "title": "Taylor Swift",
    "artist": "Taylor Swift",
    "url": "https://www.amazon.com/Taylor-Swift/dp/B0014I4KH6",
    "image": "https://images-na.ssl-images-amazon.com/images/I/61McsadO1OL.jpg",
    "thumbnail_image": "https://i.imgur.com/K3KJ3w4h.jpg"
  }
];

describe('AlbumDetailComponent', () => {
  it('renders as expected', () => {
    const rendered = renderer.create(<AlbumDetail key={albumStub[0].title} album={albumStub[0]} />).toJSON();
    expect(rendered).toBeTruthy();
    expect(rendered).toMatchSnapshot();
  });
});
