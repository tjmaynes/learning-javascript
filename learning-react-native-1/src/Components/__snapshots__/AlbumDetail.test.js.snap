// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlbumDetailComponent renders as expected 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Taylor Swift
  </Text>
</View>
`;
